src/java/org/apache/cassandra/cql3/restrictions/SimpleRestriction.java

```diff
@@ -345,11 +352,14 @@ public void addToRowFilter(RowFilter filter, IndexRegistry indexRegistry, QueryO
                 else if (operator == Operator.LIKE)
                 {
                     LikePattern pattern = LikePattern.parse(buffers.get(0));
-                    // there must be a suitable INDEX for LIKE_XXX expressions
+                    
                     RowFilter.SimpleExpression expression = filter.add(column, pattern.kind().operator(), pattern.value());
-                    indexRegistry.getBestIndexFor(expression, indexHints)
-                                 .orElseThrow(() -> invalidRequest("%s is only supported on properly indexed columns",
-                                                                   expression));
+                    Optional<Index> index = indexRegistry.getBestIndexFor(expression, indexHints);
+                    if(!index.isPresent() && !allowFiltering)
+                    {
+                        throw invalidRequest("%s is only supported on properly indexed columns or with ALLOW FILTERING",
+                                            expression);
+                    }
```

Comment:
nit: I might just go single-line `if` here. It probably won't ever be multi-line...

```
if (!index.isPresent() && !allowFiltering)
    throw invalidRequest("%s is only supported on properly indexed columns or with ALLOW FILTERING", expression);
```